src/engine.rs

```diff
@@ -32,6 +32,18 @@ struct PromptWidget {
     origin: (u16, u16),
 }
 
+enum ReedlineEvent {
+    HandleTab,
+    CtrlD, // Don't know a better name for this
+    CtrlC, // Don't know a better name for this
+    ClearScreen,
+    Enter,
+    Mouse, // Fill in details later
+    Resize(u16, u16),
+    EditInsert(EditCommand), // HACK: Special handling for insert
```

Comment:
As mentioned it's a simple hack for now, I would not like a special case for InsertCommand